src/com/ms/silverking/cloud/dht/client/gen/Variable.java

```diff
@@ -16,449 +16,507 @@
+    String n;
+
+    n = t.getTypeName();
+    if (n.startsWith("class ")) {
+      n = n.substring("class ".length());
+    }
+    return n;
+  }
+
+  /*
+  public static String getReturnTypeWrapper(Method m, String s) {
+      String	name;
+      TypeMapping	javaTypeMapping;
+
+      name = m.getReturnType().toString();
+      javaTypeMapping = javaTypeMappings.get(name);
+      if (javaTypeMapping != null) {
+          return javaTypeMapping.getSKToJavaGenerator().generate(s);
+      } else {
+          return s;
+      }
+  }
+  */
+  public static String getReturnTypeWrapper(Method m) {
+    String name;
+    TypeMapping typeMapping;
+
+    name = getTypeNameForMapping(m.getReturnType());
+    System.out.printf("getReturnTypeWrapper %s %s\n", m.getName(), name);
+    typeMapping = javaTypeMappings.get(name);
+    if (typeMapping != null) {
+      String wrapper;
+
+      wrapper = typeMapping.getJavaToSKGenerator().getReturnWrapper();
+      if (wrapper != null) {
+        return wrapper;
+      } else {
+        return "";
+      }
+    } else {
+      return "";
+    }
+  }
+
+  private String getParameterNameWrapped(Parameter p) {
+    String name;
+    TypeMapping javaTypeMapping;
+
+    name = getTypeNameForMapping(p.getParameterizedType());
+    javaTypeMapping = javaTypeMappings.get(name);
+    if (javaTypeMapping != null) {
+      return javaTypeMapping.getSKToJavaGenerator().generate(p.getName());
+    } else {
+      return p.getName();
+    }
+  }
+
+  private static String getTypedClassName(Class c) {
+    String g;
+    int i;
+
+    g = c.toGenericString();
+    i = g.indexOf(c.getName());
+    return g.substring(i);
+  }
+
+  private static String getParameterTypePackage(Parameter p) {
+    String name;
+    TypeMapping javaTypeMapping;
+
+    name = p.getParameterizedType().getTypeName();
+    //name = p.getParameterizedType().toString();
+    javaTypeMapping = javaTypeMappings.get(name);
+    //System.out.printf("\t###\t%s\t%s\n", name, javaTypeMapping);
+    if (javaTypeMapping == null) {
+      return getTypePackage(p.getType());
+    } else {
+      return "";
+    }
+  }
+
+  private static String getParameterType(Parameter p, boolean simple, String enclosingTypeSeparator) {
+    String name;
+    TypeMapping javaTypeMapping;
+
+    //name = p.getParameterizedType().toString();
+    name = p.getParameterizedType().getTypeName();
+    javaTypeMapping = javaTypeMappings.get(name);
+    //System.out.printf("\t***%s\t%s\n", p.getName(), javaTypeMapping == null ? "null" : javaTypeMapping
+    // .getExternalType());
+    if (javaTypeMapping == null) {
+      System.out.printf("%s\n", name);
+    }
+    if (javaTypeMapping == null) {
+      if (!simple) {
+        return name;
+      } else {
+        return getSimpleTypeName(p.getType(), enclosingTypeSeparator);
+        //return p.getType().getSimpleName();
+      }
+    } else {
+      return javaTypeMapping.getExternalType();
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 434:443, 455:464.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.